### PR TITLE
CNI: Add support for OpenShift and Multus

### PR DESCRIPTION
**Pull request by curtbushko**
Changes proposed in this PR:

This PR adds support for OpenShift and Multus in CNI. OpenShift uses Multus as the base CNI plugin.

- Added SecurityContextConstraints CRD so that CNI has the permissions to run on OpenShift and gave the consul-cni service account permissions to reach OpenShifts SCC CRD.
- Added a --multus flag to the CNI installer so that the flag can be used by the CNI plugin.
- When --multus flag is set, create a default Network Attachment Definition CRD for Multus to use. Note, Pods require an annotation in order for Multus to run the consul-cni plugin. Please see [Multus documentation](https://github.com/k8snetworkplumbingwg/multus-cni) for more information about how to use Multus.
- Fix: Fixed installer to rename .conf file .conflist file when consul-cni is added to a base plugin.
- Improvement: The plugin is now more consistent at updating annotations. 

How I've tested this PR:

- Manual testing of Multus on OpenShift
- Manual testing of Multus on Kind
- Acceptance and unit tests.

How I expect reviewers to test this PR:

:eyes:

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/cni-securitycontextconstraints.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: cni
+  annotations:
+    kubernetes.io/description: {{ template "consul.fullname" . }}-cni are the security context constraints required
+      to run consul-cni.
+allowHostDirVolumePlugin: true
+allowHostIPC: false
+allowHostNetwork: true
```
Comment (ishustava) [inline charts/consul/templates/cni-securitycontextconstraints.yaml:18]: Double checking: Do we need host network?

File: charts/consul/templates/cni-securitycontextconstraints.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+allowHostPID: false
+allowHostPorts: true
```
Comment (ishustava) [inline charts/consul/templates/cni-securitycontextconstraints.yaml:20]: do we need host ports?

File: charts/consul/values.yaml [hunk 3]
```diff
@@ -1960,7 +1960,20 @@ connectInject:
...
+    # CNI plugin. Instead, a NetworkAttachementDefinition CustomResourceDefinition (CRD) will be created in the helm
+    # release namespace. Following multus plugin standards, an annotation is required in order for the consul-cni plugin
+    # to be executed and for your service to be added to the Consul Service Mesh. 
+    #
+    # Add the annotation `'k8s.v1.cni.cncf.io/networks': '[{ "name":"consul-cni","namespace": "consul" }]'` to your pod
+    # to use the default installed NetworkAttachementDefinition CRD.
+    #
+    # Please refer to the [Multus Quickstart Guide](https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/quickstart.md)
+    # for more information about using multus. 
+    # @type: string
```
Comment (ishustava) [inline charts/consul/values.yaml:1974]: Beautiful!

**Review by ishustava — APPROVED** [review]
It's perfect!!! Just a couple of clarifying comments.

File: charts/consul/templates/cni-securitycontextconstraints.yaml (hunk 1 above)
Comment (curtbushko) [reply charts/consul/templates/cni-securitycontextconstraints.yaml]: Nice catch!

I just sanity tested and we do not.  They have been removed.

File: charts/consul/templates/cni-securitycontextconstraints.yaml (hunk 2 above)
Comment (curtbushko) [reply charts/consul/templates/cni-securitycontextconstraints.yaml]: I just sanity tested and we do not.  They have been removed.

**Comment by david-yu** [comment]
Thanks @curtbushko would also be possible to create a PR for how to use the CNI Plugin on Openshift here: https://www.consul.io/docs/k8s/installation/install#enable-the-consul-cni-plugin

**Review by wilkermichael — APPROVED** [review]

**Pull request merged**